src/chart/FtpHistoryChart.svelte

```diff
@@ -0,0 +1,80 @@
+<script>
+    import { onMount } from "svelte";
+    import Chart from "chart.js";
+    import Card from "../components/Card.svelte";
+    import * as ChartAnnotation from "chartjs-plugin-annotation";
+    import { ftpAverageOutputs } from "../store/store.js";
+    import moment from "moment";
+
+    const getPlotPoints = (data) => {
+        return data.map((ride) => {
+            // Functional Threshold Power (FTP) is average output over 1h in 20m tests discount by 5%
+            return {
+                y: Math.floor(ride.average * 0.95),
+                x: moment(ride.createdAt, "YYYY-MM-DD"),
+            };
+        });
+    };
+
+    const getChartData = (data) => {
+        return {
+            datasets: [
+                {
+                    borderColor: "#00cc00",
+                    label: "FTP Tests",
+                    data: getPlotPoints(data),
+                    fill: false,
+                    lineTension: 0,
+                },
+            ],
+        };
+    };
+
+    let ftpHistoryChart;
+
+    let config = {
+        type: "line",
+        plugins: [ChartAnnotation],
+        data: getChartData($ftpAverageOutputs),
+        options: {
+            maintainAspectRatio: false,
+            responsive: true,
+            scales: {
+                xAxes: [
+                    {
+                        type: "time",
+                        distribution: "series",
+                        bounds: "data",
+                        time: {
+                            unit: "day",
+                            tooltipFormat: "MMM DD YYYY",
+                        },
+                    },
+                ],
+            },
+        },
+    };
+
+    onMount(async () => {
+        let ctx = document.getElementById("ftpHistoryChart");
+        ftpHistoryChart = new Chart(ctx, config);
+
+        ftpAverageOutputs.subscribe((value) => {
+            ftpHistoryChart.data = getChartData(value);
+            ftpHistoryChart.update();
+        });
+    });
+</script>
+
+<style>
+    div {
+        min-height: 90vh;
+    }
+</style>
+
+{#if $ftpAverageOutputs.length > 0}
```

Comment:
Sorry, I was wrong... ran it and this results in the following console error:
```Chart.js:7739 Uncaught (in promise) TypeError: Cannot read property 'length' of null
    at Object.acquireContext (Chart.js:7739)
    at Chart.construct (Chart.js:9307)
    at new Chart (Chart.js:9294)
    at FtpHistoryChart.svelte:60
    at run (index.mjs:18)
    at Array.map (<anonymous>)
    at index.mjs:1379
    at flush (index.mjs:715)
```
because line 59 runs when the component mounts even if the html here doesn't get loaded. Unfortunately, this means importing `ftpAverageOutputs` into `MainContent.svelte` and wrapping `<FtpHistoryChart />` in this same logic.

Reply:
ha that is what I initially did and tested, but saw your comment of putting here. Actually is my bad as when I refactored to within this card I didn't test it 🤦

Reply:
I just forced pushed to remove that commit now this should work as expected. I tested with modifying my data to remove the test rides.

https://github.com/jasonlutterloh/metrics-peloton/commit/370d0d7e8335e2a252471abc20cceef29b46be9b